### PR TITLE
chapter 1 finished

### PR DESCRIPTION
**Pull request by volker48**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -490,7 +490,10 @@ Implement a function that returns the last digit of a given number.
   whether it works for you!
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit n = 
+  let pos = abs n
+      in mod pos 10
```
Comment (vrom911) [inline src/Chapter1.hs:496]: Nice usage of `let-in` 💯

File: src/Chapter1.hs [hunk 2]
```diff
@@ -520,7 +523,7 @@ branches because it is an expression and it must always return some value.
   satisfying the check will be returned and, therefore, evaluated.
 -}
 closestToZero :: Int -> Int -> Int
-closestToZero x y = error "closestToZero: not implemented!"
+closestToZero x y = if (abs x) < abs(y) then x else y
```
Comment (vrom911) [inline src/Chapter1.hs]: In Haskell, you don't need parenthesis in both of these cases:

```suggestion
closestToZero x y = if abs x < abs y then x else y
```

Also, functions and arguments are always separated with spaces even when arguments are in parenthesis 👾

File: src/Chapter1.hs [hunk 3]
```diff
@@ -631,8 +645,13 @@ Implement a function that returns the sum of the last two digits of a number.
...
 specifying complex expressions.
 -}
-
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n =
+  let pos = abs n
+      lastTwo = mod pos 100
+      final = mod lastTwo 10
+      secondLast = div lastTwo 10
```
Comment (vrom911) [inline src/Chapter1.hs]: That is a wonderful solution! 👏🏼  You correctly noticed that it is the `div` and `mod`, cool 😎 

One hint to make your solution even shorter: the standard library has the [`divMod`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:divMod) function, that actually combines inside both `div` and `mod`. And this is exactly what you use!.

So you could write it this way:

```haskell
(x, y) = divMod lastTwo 10
```

You can see how we could pattern match on the pair 🙂

**Review by vrom911 — APPROVED** [review]
Awesome work on solving the challenges! ⭐ 
Congrats with finishing Chapter One 🥳 
And well done with the Advanced Task :medal_sports: 

I left my review and some comments. Hope you find them helpful :slightly_smiling_face:

**Comment by volker48** [comment]
@vrom911 Thank you for taking the time to review this and for putting this project together.

**Pull request merged**